### PR TITLE
CXXCBC-80: do not crash when opening non-existent bucket

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

**Review by jrawsthorne — DISMISSED** [review]

**Comment by avsej** [comment]
test this please

File: couchbase/io/http_session.hxx [hunk 1]
```diff
@@ -405,9 +410,10 @@ class http_session : public std::enable_shared_from_this<http_session>
 
     void do_read()
     {
-        if (stopped_) {
+        if (stopped_ || reading_ || !stream_->is_open()) {
```
Comment (brett19) [inline couchbase/io/http_session.hxx:413]: Little suspicious as this doesn't appear related to unknown bucket bootstrap crashing.

**Review by brett19 — APPROVED** [review]

File: couchbase/io/http_session.hxx (hunk 1 above)
Comment (avsej) [reply couchbase/io/http_session.hxx:413]: I agree with you.
